ludwig/modules/embedding_modules.py

```diff
@@ -79,9 +79,8 @@ def embedding_matrix(
 
     elif representation == 'sparse':
         embedding_size = vocab_size
-        embeddings = torch.tensor(
-            get_initializer('identity')([vocab_size, embedding_size]),
-            requires_grad=False)
+        embeddings = get_initializer('identity')(
+            [vocab_size, embedding_size]).clone().detach().requires_grad_(False)
```

Comment:
curious here, why are clone and detach needed and why this different way of setting requires grad to false?

Reply:
This was a warning/recommendation from pytorch:

```
UserWarning: To copy construct from a tensor, it is recommended to use source
Tensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True),
 rather than torch.tensor(sourceTensor)
```

It looks like it's slightly more efficient, though also less readable:
https://stackoverflow.com/questions/55266154/pytorch-preferred-way-to-copy-a-tensor

Reply:
I like the second option. Yeah it's not clear why in the original code we had to copy the tensor returned by get_initializer(...) into another tensor.